File: 7_Function_Overloading.cpp
#include <bits/stdc++.h>
using namespace std;

class Printer
{
public:
    void print(int num)
    {
        cout << "The printing integer: " << num << endl;
    }

    void print(float num)
    {
        cout << "The printing float: " << num << endl;
    }

    void print(string s)
    {
        cout << "The printing string: " << s << endl;
    }
};

int main()
{
    Printer printer;

    printer.print(20);
    printer.print(3.14f);
    printer.print("Hello,I'm Turza");

    return 0;
}

File: 7_Function_Overloading.py
# python doesn't have method overloading
# but it can be use by below procedure  

class Printer:
    def print_value(self, value):
        if isinstance(value, int):
            print("Printing integer:", value)
        elif isinstance(value, float):
            print("Printing float:", value)
        elif isinstance(value, str):
            print("Printing string:", value)

def main():
    printer = Printer()

    printer.print_value(20)
    printer.print_value(3.14)
    printer.print_value("Hello,I'm Turza")

if __name__ == "__main__":
    main()
